rice-middleware/impl/src/main/java/org/kuali/rice/kew/routelog/web/RouteLogAction.java

```diff
@@ -155,7 +157,12 @@ private ActionRequestValue switchActionRequestPositionIfPrimaryDelegatePresent(
     	 */
     	
     	if (!actionRequest.isRoleRequest()) {
-    		List<ActionRequestValue> primaryDelegateRequests = actionRequest.getPrimaryDelegateRequests();
+            /**
+             * KULRICE-14183 : Detaching the action request object so that re-ordering of action requests for primary delegates does not get incorrectly persisted
+             */
+            EntityManager entityManager = GlobalResourceLoader.getService("rice.kew.entityManager");
```

Comment:
No - we would not.  Any JPA-specific code should (generally) only be in DAO classes.  It's probably the case that, if we need to alter the objects for display purposes, that we need to either clone into new objects or use the DTO classes used by services.